### PR TITLE
Add support for incoming application control

### PR DESCRIPTION
**Pull request by rwalczyna**
This change is needed to provide support for OnAppControlReceived callback.
As there is no way to pass object from embedder to plugins without changing plugins API, I have decided to create internal App Control Channel inside engine. I am open for discussion if there is any easier solution for this case.

Corresponding engine change: https://github.com/flutter-tizen/engine/pull/154

Dart part of the plugin is currently in internal f-project repository.

I wanted to use most recent code of embedder and engine and I had to put this PR after #171, so #171 should land first.

### CODE REVIEW

**Comment by WonyoungChoi** [comment]
@rwalczyna please rebase this PR first.

File: embedding/cpp/flutter_app.cc [hunk 1]
```diff
@@ -85,6 +92,11 @@ void FlutterApp::OnRegionFormatChanged(app_event_info_h event_info) {
   FlutterDesktopNotifyLocaleChange(handle_);
 }
 
+void FlutterApp::OnAppControlReceived(app_control_h app_control) {
+  TizenLog::Debug("OnAppControlReceived");
+  FlutterDesktopNotifyAppControl(handle_, app_control);
+}
```
Comment (swift-kim) [inline embedding/cpp/flutter_app.cc:98]: - Please move these lines between `FlutterApp::OnTerminate` and `FlutterApp::OnLowMemory` to match the declaration order. Please apply the same thing to the C# embedding for consistency.
- We shouldn't need the debug log.
- Also please add<p>
  ```cpp
  assert(IsRunning());
  ```
  for consistency.

**Comment by bbrto21** [comment]
@rwalczyna 
Question : after all of the things related to this, will we have a structure similar to supporting ```intent``` in Android?
https://flutter.dev/docs/get-started/flutter-for/android-devs#how-do-i-handle-incoming-intents-from-external-applications-in-flutter

File: embedding/csharp/FlutterApplication.cs [hunk 2]
```diff
@@ -203,6 +203,15 @@ protected override void OnRegionFormatChanged(RegionFormatChangedEventArgs e)
             FlutterDesktopNotifyLocaleChange(Handle);
         }
 
+        protected override void OnAppControlReceived(AppControlReceivedEventArgs e)
```
[the remark this answers is not in the document]
Comment (pkosko) [reply embedding/csharp/FlutterApplication.cs:178]: Done https://github.com/flutter-tizen/flutter-tizen/pull/174/files#diff-702ac9e29a07661f329be027af02b4c68b568fab43ccafe3d09fa20ee5a652a1R119

**Comment by swift-kim** [comment]
I published a new [engine release](https://github.com/flutter-tizen/engine/releases) which includes the change https://github.com/flutter-tizen/engine/pull/154.

Please update the engine hash (`bin/internal/engine.version`) to `7ffeb878c5d4d703968defe9fcbe2fed294c1a81`.

**Comment by pkosko** [comment]
> 
> 
> @rwalczyna
> Question : after all of the things related to this, will we have a structure similar to supporting `intent` in Android?
> https://flutter.dev/docs/get-started/flutter-for/android-devs#how-do-i-handle-incoming-intents-from-external-applications-in-flutter

It is an implementation for https://docs.tizen.org/application/web/guides/app-management/app-controls/, but I believe that it is a similar mechanism as in quoted link.

File: embedding/csharp/FlutterServiceApplication.cs [hunk 3]
```diff
@@ -116,6 +116,13 @@ protected override void OnRegionFormatChanged(RegionFormatChangedEventArgs e)
             FlutterDesktopNotifyLocaleChange(Handle);
         }
 
+        protected override void OnAppControlReceived(AppControlReceivedEventArgs e)
+        {
+            Debug.Assert(Handle);
+
+            FlutterDesktopNotifyAppControl(Handle, e.ReceivedAppControl.SafeAppControlHandle);
+        }
+
```
[the remark this answers is not in the document]
Comment (swift-kim) [reply embedding/csharp/FlutterServiceApplication.cs]: Please apply the same thing to `FlutterApplication.cs`.

[the remark this answers is not in the document]
Comment (pkosko) [reply embedding/csharp/FlutterServiceApplication.cs]: Sorry, I've missed that. Now it is fixed also in FlutterApplication.cs

**Review by swift-kim — APPROVED** [review]
Thank you!

**Pull request merged**